man/man3/ConfigCDRom.3
.TH "ConfigCDRom" 3 "Thu Aug 23 2012" "Version 1.0" "FS-UAE Gui for Linux OS" \" -*- nroff -*-
.ad l
.nh
.SH NAME
ConfigCDRom \- 
.PP
On the success of the A500, Commodore launches in the production of alternative projects, that are based on the \fBAmiga\fP\&. The 'CDTV', of 1990, is designed as a household appliance for multimedia entertainment, and is basically an A500 with a CD-Rom, in a chassis similar to that of a VCR\&. 
.br
 The intention of the Commodore, the CDTV must implement the convergence of television and computers, allowing the user to play games, watch movies and listen to music with the same equipment, the most advanced user can then expand the purchasing CDTV keyboard, floppy disk drives, and mouse and turn it into an actual computer\&.
.br
 AmigaCD32 (1993), is instead a console dedicated to video games, equipped with 68020 processor with 32-bit architecture and is based on the use of a CD-Rom integrated\&. 
.br
 
.br
 This class is responsible for the Cd-Rom Configuration management\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <configcdrom\&.h>\fP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBConfigCDRom\fP ()"
.br
.ti -1c
.RI "string \fBgetCDRomDrive0ConfigString\fP ()"
.br
.RI "\fIRepresents the string of cdrom_drive_0 to be written in Config File\&. \fP"
.ti -1c
.RI "string \fBgetCDRomDrive0String\fP ()"
.br
.RI "\fIRepresent the effective value of cdrom_drive_0\&. \fP"
.ti -1c
.RI "int \fBgetCDRomImageSize\fP ()"
.br
.RI "\fIRepresent the image of a CD-Rom\&. \fP"
.ti -1c
.RI "string \fBgetCDRomImageAt\fP (int position)"
.br
.RI "\fIRepresent the position of a CD-Rom Image\&. \fP"
.ti -1c
.RI "void \fBpushBackCDRomImage\fP (string s)"
.br
.ti -1c
.RI "void \fBerasCDRomImageAt\fP (int position)"
.br
.ti -1c
.RI "int \fBsetParameter\fP (string parameter, string value)"
.br
.ti -1c
.RI "void \fBsetToDefaultConfiguration\fP ()"
.br
.ti -1c
.RI "bool \fBhasParameter\fP (string parameterName)"
.br
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "string \fBcdrom_drive_0\fP"
.br
.RI "\fIYou specify paths for CD-ROM images in cdrom_drive_0\&. \fP"
.ti -1c
.RI "vector< string > \fBcdrom_image\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
On the success of the A500, Commodore launches in the production of alternative projects, that are based on the \fBAmiga\fP\&. The 'CDTV', of 1990, is designed as a household appliance for multimedia entertainment, and is basically an A500 with a CD-Rom, in a chassis similar to that of a VCR\&. 
.br
 The intention of the Commodore, the CDTV must implement the convergence of television and computers, allowing the user to play games, watch movies and listen to music with the same equipment, the most advanced user can then expand the purchasing CDTV keyboard, floppy disk drives, and mouse and turn it into an actual computer\&.
.br
 AmigaCD32 (1993), is instead a console dedicated to video games, equipped with 68020 processor with 32-bit architecture and is based on the use of a CD-Rom integrated\&. 
.br
 
.br
 This class is responsible for the Cd-Rom Configuration management\&. 

In this class are defined all the variables for the correct management of CDROM configuration's tab\&.
.br
 The use of the Cd-Rom is only permitted if the \fBAmiga\fP model selected si CDTV or CD32\&.
.br
 In all other cases the Cd-Rom Configuration tabs appears disabled\&. 
.SH "Constructor & Destructor Documentation"
.PP 
.SS "\fBConfigCDRom::ConfigCDRom\fP ()"
.SH "Member Function Documentation"
.PP 
.SS "void \fBConfigCDRom::erasCDRomImageAt\fP (intposition)"
.SS "string \fBConfigCDRom::getCDRomDrive0ConfigString\fP ()"
.PP
Represents the string of cdrom_drive_0 to be written in Config File\&. CUE/BIN and ISO images are supported\&.
.br
 cdrom_drive_0 = path/to/cue_or_bin 
.SS "string \fBConfigCDRom::getCDRomDrive0String\fP ()"
.PP
Represent the effective value of cdrom_drive_0\&. 
.SS "string \fBConfigCDRom::getCDRomImageAt\fP (intposition)"
.PP
Represent the position of a CD-Rom Image\&. 
.SS "int \fBConfigCDRom::getCDRomImageSize\fP ()"
.PP
Represent the image of a CD-Rom\&. 
.SS "bool \fBConfigCDRom::hasParameter\fP (stringparameterName)"
.SS "void \fBConfigCDRom::pushBackCDRomImage\fP (strings)"
.SS "int \fBConfigCDRom::setParameter\fP (stringparameter, stringvalue)"
.SS "void \fBConfigCDRom::setToDefaultConfiguration\fP ()"
.SH "Member Data Documentation"
.PP 
.SS "string \fBConfigCDRom::cdrom_drive_0\fP\fC [private]\fP"
.PP
You specify paths for CD-ROM images in cdrom_drive_0\&. 
.SS "vector<string> \fBConfigCDRom::cdrom_image\fP\fC [private]\fP"

.SH "Author"
.PP 
Generated automatically by Doxygen for FS-UAE Gui for Linux OS from the source code\&.


configcdrom.cpp
/*! \file configcdrom.cpp
 configcdrom.cpp

 This program is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.

 This program is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.

 You should have received a copy of the GNU General Public License
 along with this program; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 Written by Riva Alessandro Mario, alessandromario.riva@gmail.com
 and Sanvito Davide, dsanvito90@gmail.com. */
#include "configcdrom.h"

const string DEFAULTCDROMEDRIVE0 = "";

ConfigCDRom::ConfigCDRom()
{
    setToDefaultConfiguration();
}

string ConfigCDRom::getCDRomDrive0ConfigString()
{
    if (cdrom_drive_0.compare(DEFAULTCDROMEDRIVE0)==0){return "";}
    else {return "cdrom_drive_0 = " + cdrom_drive_0;}
}

string ConfigCDRom::getCDRomDrive0String()
{
    return cdrom_drive_0;
}

int ConfigCDRom::getCDRomImageSize()
{
    return cdrom_image.size();
}

string ConfigCDRom::getCDRomImageAt(int position)
{
    if (position>=(int)cdrom_image.size()){return "";}
    return cdrom_image.at(position);
}

void ConfigCDRom::pushBackCDRomImage(string s)
{
    cdrom_image.push_back(s);
}

void ConfigCDRom::eraseCDRomImageAt(int position)
{
    if (position>=(int)cdrom_image.size()){return;}
    cdrom_image.erase(cdrom_image.begin()+ position);
}

int ConfigCDRom::setParameter(string parameter, string value)
{
    if(parameter.compare("cdrom_drive_0")==0){
        cdrom_drive_0=value;
    } else if((parameter.substr(0,parameter.length()-1).compare("cdrom_image")==0 || parameter.substr(0,parameter.length()-2).compare("cdrom_image")==0)&& cdrom_image.size()<20){
        cdrom_image.push_back(value);
    }
    return 0;
}

void ConfigCDRom::setToDefaultConfiguration()
{
    cdrom_drive_0=DEFAULTCDROMEDRIVE0;
    cdrom_image.clear();
}

bool ConfigCDRom::hasParameter(string parameterName)
{
    return (parameterName.substr(0,parameterName.length()-1).compare("cdrom_drive_")==0)||(parameterName.substr(0,parameterName.length()-1).compare("cdrom_image_")==0)||(parameterName.substr(0,parameterName.length()-2).compare("cdrom_image_")==0);
}
